Add tracking objects for Microsoft Forms/-Stream/Slideshare/Youtube/Maps (#158)

okroener opened the pull request:
Hello @jonaseberle ,

I have just added some support for Microsoft Forms cookies. Others will follow.

Thanks for your great work on this extension.

jonaseberle commented:
Thank you @okroener! 

Please disregard the failing CGL tests in master. We have an incompatibility between php-cs-fixer and the newest symfony/event-dispatcher with the current TYPO3 dev-master and we need to investigate.

We would be happy to have the new TrackingObjects in https://github.com/dmind-gmbh/extension-cookieman/blob/master/Configuration/TypoScript/Example/setup.typoscript, too. So please feel free to re-add it :)

Could you add your new strings to https://github.com/dmind-gmbh/extension-cookieman/blob/master/Resources/Private/Language/locallang.xlf, please? That will make them available for translators on https://crowdin.com/project/typo3-extension-cookieman.

okroener commented:
Hello @jonaseberle, thank you very much for your quick reply.

Thanks for metioning the failing CGL tests, I have seen already that they work fine in other branches like 10lts.

I am currently adding some more tracking objects and once I have finished that, I will take care of the translation and also update ../Example/setup.typoscript

Then I will perform my pull request again.

okroener commented:
Hello @jonaseberle,

I have now added futher tracking objects, changed the example and the translations as well.

There is only one thing that I am not sure about at the moment. E.g. if there are two tracking objects with the same cookie name, e.g.

Slideshare: UID and
xyz: UID

At the moment I can't see a way to have different translations for both different cookies since the translation files pick the cookie name, but not TackingObject.cookie_name?

Another time thanks for the brilliant work!

jonaseberle commented:
Hi @okroener thanks again!

Regarding Duplicate tracking object names:

It displays `trackingobject.cookie_name` and falls back to `cookie_name` (https://github.com/dmind-gmbh/extension-cookieman/blob/master/Resources/Private/Partials/TableRowsByGroup.html#L8)

You can just call your trackingObject differently. So following your example, let's say `UID` is already defined and you want to add a second one with the same name:

```
# TypoScript
plugin.tx_cookieman.settings.trackingObjects {
    Slideshare {
        show {
            Slideshare_UID {
                duration = 1
                durationUnit = year
                type = cookie_http+html
                provider = Slideshare
# this makes sure the cookie "UID" will be deleted when unconsenting, although we called it Slideshare_UID internally.
                htmlCookieRemovalPattern = ^UID$
            }
        }
    }
}
```

```
# .xlf
<trans-unit id="trackingobject.slideshare_UID" resname="trackingobject.slideshare_UID">
  <source>UID</source>
</trans-unit>
<trans-unit id="trackingobject.slideshare_UID.desc" resname="trackingobject.slideshare_UID.desc">
  <source>Used for ...</source>
</trans-unit>
```

jonaseberle commented on Configuration/TypoScript/TrackingObjects/MicrosoftStream.typoscript:
```diff
@@ -0,0 +1,50 @@
+            Signature_Api {
+                duration =
+                durationUnit = session
+                type = cookie_http
```
Are these really just HTTP cookies? If it's set via JavaScript, it would be a cookie_http+html. That would allow us to delete it when consent is revoked later.
Please confirm also for MicrosoftForms.

jonaseberle commented:
I am blown away @okroener! You also included Slideshare, Maps and Youtube! Splendid, thank you!

jonaseberle reviewed (changes requested):
Just the question if it's HTTP cookies should be answered. I am happy to merge then.

okroener commented:
Hello @jonaseberle,

thank you very much for your reply and the clarification of the translations. I should have looked into the template first, before asking that question :-)

Regarding the cookies in MicrosoftForms/-Stream I have checked that again and have found out, that in Stream the cookies are all HttpOnly, but in Forms there are different types of cookies, like:

```
...
            __RequestVerificationToken {
                duration =
                durationUnit = session
                type = cookie_http
                provider = Microsoft
            }
            
            AADNonce\.forms {
                duration =
                durationUnit = session
                type = cookie_http
                provider = Microsoft
            }

            DcLcid {
                duration = 3
                durationUnit = months
                type = cookie_http
                provider = Microsoft
            }

            MSFPC {
                duration = 1
                durationUnit = year
                type = cookie_http+html
                provider = Microsoft
            }

            MUID {
                duration = 1
                durationUnit = year
                type = cookie_http+html
                provider = Microsoft
            }

```
Keep up the great work!

jonaseberle commented:
Thank you @okroener!

okroener commented:
You are welcome @jonaseberle and thanks for triggering the translation.

At the moment I am adding more tracking objects in my local dev branch. I will make a pull request when I have completed that list in order to reduce the merging efforts.

Thanks!
